doc/conf.py

```diff
@@ -12,6 +12,16 @@
 # serve to show the default.
 
 import sys
+from unittest.mock import MagicMock
```

Comment:
This breaks local doc compilation. I was wondering, it is needed at all since you disabled [installing extension when compiling from RtD](https://github.com/hyperspy/hyperspy/pull/1155/files#diff-2eeaed663bd0d25b7e608891384b7298R65)?

Even if not for the original purpose, it may actually be useful (with a bit of tweaking) for getting the API docs also in RtD (they are currently broken because of the inability to install the libraries in `MOCK_MODULES`.)

What do you think @sem-geologist?